src/passes/MinifyImportsAndExports.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2018 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Minifies import and export names, renaming them to short versions,
+// and prints out a mapping to the new short versions. That mapping
+// can then be used to minify the JS calling the wasm, together enabling
+// minification of the identifiers on the JS/wasm boundary.
+//
+// For example, this may minify
+//   (import "env" "longname" (func $internal))
+// to
+//   (import "env" "a" (func $internal))
+// "a" is the minified name (note that we only minify the base,
+// not the module).
+//
+
+#include <map>
+#include <string>
+#include <unordered_set>
+
+#include <wasm.h>
+#include <pass.h>
+#include <shared-constants.h>
+#include <asmjs/shared-constants.h>
+#include <ir/import-utils.h>
+#include <ir/module-utils.h>
+
+namespace wasm {
+
+struct MinifyImportsAndExports : public Pass {
+
+  // Generates minified names that are valid in JS.
+  // Names are computed lazily.
+  class MinifiedNames {
+  public:
+    MinifiedNames() {
+      // Reserved words in JS up to size 4 - size 5 and above would mean we use an astronomical
+      // number of symbols, which is not realistic anyhow.
+      reserved.insert("do");
+      reserved.insert("if");
+      reserved.insert("in");
```

Comment:
Hi @kripken. Shouldn't this also reserve `null`, `eval` and `true` keywords?

Reply:
Hmm yeah, the list is incomplete - I just copied it from the emscripten js minifier code. Which is incomplete too I guess ;)

I think in practice, to get to 4 letters is very unlikely. With three letters you can get to 221,184 minified ids. But, would be worth adding the rest of the words too, I agree.